### ISSUE TITLE
Getting TypeError upon calling extrema_bounding(G, compute="eccentricities") in distance_measures.py

### ISSUE THREAD

**Issue by jbbeltran** [networkx/networkx#5402]
I tried using the ``extrema_bounding`` function from ``networkx.algorithms.distance_measures`` to compute eccentricity values for all nodes in a Graph. The documentation does not present the "eccentricities" option for the ``compute`` keyword argument but it is available in the source code so I went ahead and tried it. The exact function call I did was ``extrema_bounding(G, compute="eccentricities")``.

Looking at the code, I believe the problem is in lines 152 and 153

```python
...
        elif compute == "eccentricities":
            ruled_out = {}
```

I believe the intention is to make ``ruled_out`` an empty set (``set([])``), but ``{}`` is interpreted as an empty dictionary.

### Current Behavior
Outputs a ``TypeError: cannot convert dictionary update sequence element #0 to a sequence``

### Expected Behavior
The function should run without errors.

### Steps to Reproduce

Running the following snippet should reproduce the bug:
```python
from networkx.generators.social import karate_club_graph
from networkx.algorithms.distance_measures import extrema_bounding

G = karate_club_graph()
ecc = extrema_bounding(G, compute="eccentricities")
```

### Environment

Python version: Python 3.8.1
NetworkX version: networkx 2.7.1

**Comment by z3y50n** [comment]
I can reproduce this error and from what I see the `eccentricities` option is not listed in the doc string as well.

Also if `compute` is not one of the five values `"diameter", "radius", "periphery", "center", "eccentricities"` it will throw an `UnboundLocalError` on the next line since `ruled_out` variable won't be initialized. https://github.com/networkx/networkx/blob/6a0b4faf09ec9d3d40ad93e2ec9b431d6bab5dc4/networkx/algorithms/distance_measures.py#L155

**Comment by rossbar** [comment]
Looking at `extrema_bounding` more carefully, I wonder if it was ever intended to be called directly? It seems like it's mostly used internally for the supported measures mentioned in the docstring and `extrema_bounding` itself is untested!

**Comment by jbbeltran** [comment]
Thanks to the contributors for your hard work! I tried to use ``extrema_bounding`` to compute for the eccentricities of _all_ nodes of a Graph (not just diameter or center), as the current ``eccentricity`` function runs in a brute-force manner, performing a single-source shortest path from every node. I also found the original source code (C++) of the author/s of the papers listed in the docstring so I'm looking into using that as well.

Thank you again.
